File: microservices/temperature-processing/src/main/java/com/algasensors/temperature/processing/infra/messaging/dto/TemperatureMessage.java
package com.algasensors.temperature.processing.infra.messaging.dto;

import java.math.BigDecimal;
import java.time.Instant;

public record TemperatureMessage(
        String messageId,
        String sensorId,
        BigDecimal temperature,
        String unit,
        Instant occurredAt
) {}

File: microservices/sensor-simulator/src/main/kotlin/com/algasensors/sensor/dto/TemperatureMessage.kt
package com.algasensors.sensor.dto

import com.fasterxml.jackson.annotation.JsonProperty
import java.math.BigDecimal
import java.time.Instant

data class TemperatureMessage(
    @JsonProperty("messageId") val messageId: String,
    @JsonProperty("sensorId") val sensorId: String,
    @JsonProperty("temperature") val temperature: BigDecimal,
    @JsonProperty(value = "unit") val unit: String,
    @JsonProperty("occurredAt") val occurredAt: Instant
)